File: dart/dojo/kata6.dart
// https://www.codewars.com/kata/55fd2d567d94ac3bc9000064/train/dart

//              1
//           3     5
//        7     9    11
//    13    15    17    19
// 21    23    25    27    29
// ...

int rowSumOddNumbers(int n) {
  var gaussiansum =
      List.generate(n, (i) => ++i).reduce((value, element) => value + element);
  var solution = List.generate(gaussiansum, (index) => index + index + 1)
      .reversed
      .take(n)
      .reduce((value, element) => value + element);
  print(solution);
  return solution;
}

// int rowSumOddNumbers(int n) => n * n * n;

main(List<String> args) {
  assert(rowSumOddNumbers(2) == 8);
  assert(rowSumOddNumbers(3) == 27);
  assert(rowSumOddNumbers(42) == 74088);
}

File: kotlin/dojo/kata6.kt
// https://www.codewars.com/kata/55cbc3586671f6aa070000fb/train/kotlin

// ex. base: 6; factor: 2 and 3
fun checkForFactor(base: Int, factor: Int): Boolean {

    val a = base % factor == 0
    println(a)
    return a
}

fun main() {
    checkForFactor(10, 2)
}
